core/logger/zap_disk_logging.go

```diff
@@ -0,0 +1,88 @@
+package logger
+
+import (
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/utils"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+var diskPollInterval = 1 * time.Minute
+
+type zapDiskPollConfig struct {
+	stop     func()
+	pollChan <-chan time.Time
+}
+
+func newDiskPollConfig(interval time.Duration) zapDiskPollConfig {
+	ticker := time.NewTicker(utils.WithJitter(interval))
+
+	return zapDiskPollConfig{
+		pollChan: ticker.C,
+		stop:     ticker.Stop,
+	}
+}
+
+func newDiskCore(cfg ZapLoggerConfig) (zapcore.Core, error) {
+	availableSpace, err := cfg.diskStats.AvailableSpace(cfg.local.Dir)
+	if err != nil || availableSpace < cfg.local.RequiredDiskSpace {
+		// Won't log to disk if the directory is not found or there's not enough disk space
+		cfg.diskLogLevel.SetLevel(zapcore.FatalLevel + 1)
```

Comment:
Fatal is the highest, so this was meant to prevent writing anything. Maybe it should be a constant though e.g.
```
const disabledLevel = zapcore.FatalLevel + 1
```

Reply:
this disables the logger, will add that constant.